### PR TITLE
[WIP] Refactor Preview

### PR DESCRIPTION
**Pull request by jasonLaster**
Associated Issue: #3053

### Summary of Changes

* stop getting variables from pause data
* move logic to an ast selection action
* stop using hidden expressions

### CODE REVIEW

File: src/components/Editor/index.js [hunk 1]
```diff
@@ -384,55 +385,29 @@ class Editor extends PureComponent {
...
       return;
     }
 
-    if (selectedToken) {
-      selectedToken.classList.remove("selected-token");
-      this.setState({ selectedToken: null, selectedExpression: null });
-    }
+    const location = getTokenLocation(this.editor.codeMirror, token);
+    setSelection(tokenText, location);
+    console.log("selection", token, tokenText, location);
```
Comment (bomsy) [inline src/components/Editor/index.js]: rouge `console.log`

File: src/utils/pause.js [hunk 2]
```diff
@@ -6,7 +6,7 @@ export function updateFrameLocations(
   frames: Frame[],
   sourceMaps: any
 ): Promise<Frame[]> {
-  if (!frames || frames.length == 0) {
+  if (true || !sourceMaps || !frames || frames.length == 0) {
```
Comment (bomsy) [inline src/utils/pause.js]: i think the `true` should not be there

File: src/reducers/ast.js [hunk 3]
```diff
@@ -44,6 +45,28 @@ function update(
...
+
+      if (!action.value) {
+        return state.set("selection", null);
+      }
+
+      const { expression, location, result } = action.value;
+      return state.set("selection", {
+        expression,
+        location,
+        result
```
Comment (bomsy) [inline src/reducers/ast.js]: do we need to make sure `updating` is false here?

Comment (jasonLaster) [inline src/reducers/ast.js]: hmm, i guess i should. was being cute :)

**Comment by bomsy** [comment]
looks good! just tested , it works nicely!
 
There seems to be some lines which should be out of scope which seem to be still on when we hover.

**Pull request merged**